### PR TITLE
Add VGC 2019 formats

### PR DESCRIPTION
**Pull request by HoeenCoder**
Not 100% done but PRed now as per TI's request.

https://www.pokemon.com/us/pokemon-news/get-ready-for-a-legendary-video-game-championships-season/

Your Time needs to be increased to 7 minutes (up from 5) in all formats.
~~Teambuilder is also trying to validate for "gen7vgc2019" rather than the actual format ID, which I am looking into currently.~~

### CODE REVIEW

**Comment by Marty-D** [comment]
I assume the client has code to ignore things in parentheses, such as (suspect test), for validation purposes.

Also as far as I can tell it's not Dragon Ascent that's banned, but Rayquaza + Dragon Ascent.

**Comment by HoeenCoder** [comment]
Would you suggest removing the ()s @Marty-D? I suppose build indexes would also need a tweak to get these new formats working since atm its expecting gen#vgc https://github.com/Zarel/Pokemon-Showdown-Client/blob/master/build-tools/build-indexes#L334

**Comment by TheImmortal** [comment]
Why does the client even validate in the regular format as opposed to the suspect test? I don't see the benefit.

**Comment by Marty-D** [comment]
No idea, but I'd use a hyphen or colon or whatever meanwhile.

The build index would probably be fine? It's still gen7 and it's still vgc, so what's the issue @HoeenCoder?

**Comment by SadisticMystic** [comment]
Based on what I've seen of 3DS format regulation files, they have a block of bitflags for banned species, another block for banned items, and a third block for banned moves. I don't think the syntax they use is capable of expressing complex bans such as 'Rayquaza + Dragon Ascent' at all, so my suspicion (which can't be confirmed until the download-rules file has been released) is that they'll end up banning Dragon Ascent outright, including on Smeargle, during phases 1 and 2.

**Comment by Zarel** [comment]
The client ignores things in parentheses so that they're in the same bucket for teambuilder reasons.

Specifically, so "UU" and "UU (suspect test)" show the same list of teams.

**Comment by Zarel** [comment]
In this case, I wouldn't use parentheses for the three formats here.

**Comment by DaWoblefet** [comment]
Seconding SadisticMystic that typically download rules aren't good with complex bans, I would ban Dragon Ascent outright until we see otherwise. Also while we're here, just want to reiterate that the your time system still has some flaws that need to be worked out (see: https://github.com/Zarel/Pokemon-Showdown/issues/4547). Increasing to 7 minute Your Time atm really doesn't change anything as far as how the timer is expected to play out.

File: config/formats.js [hunk 1]
```diff
@@ -271,6 +271,93 @@ let Formats = [
...
+		onValidateTeam: function (team) {
+			const legends = ['Mewtwo', 'Lugia', 'Ho-Oh', 'Kyogre', 'Groudon', 'Rayquaza', 'Dialga', 'Palkia', 'Giratina', 'Reshiram', 'Zekrom', 'Kyurem', 'Xerneas', 'Yveltal', 'Zygarde', 'Cosmog', 'Cosmoem', 'Solgaleo', 'Lunala', 'Necrozma'];
+			let n = 0;
+			let problems = [];
+			for (const set of team) {
+				const baseSpecies = this.getTemplate(set.species).baseSpecies;
+				if (legends.includes(baseSpecies)) n++;
+				if (n > 2) {
+					problems.push(`You can only use up to two legendary Pok\u00E9mon.`);
+					n = -6; // Prevent the message from being listed more than once.
```
Comment (urkerab) [inline config/formats.js]: Perhaps put the legends count check at the end? (You would also have the option of suffixing `(You have ${n} legendary Pok\u00E9mon.)`.)

**Comment by TheImmortal** [comment]
Changed the name again. And how's that @urkerab?

**Comment by TheImmortal** [comment]
Timer stuff can be changed later I guess.

**Pull request merged**